### ISSUE TITLE
Pricebook error on UnitOfWork test

### ISSUE THREAD

**Issue by renatoliveira** [financialforcedev/fflib-apex-common#169]
First let me say that I'm not using the framework, so I'm not sure if this is related to my org or the project itself. I'm just using the UnitOfWork class (I'm on an existing project, so there's a lot of legacy codebase to migrate to a new framework).

I'm getting the following error when running the test class for the unit of work:

 `System.DmlException: Insert failed. First exception on row 0; first error: DUPLICATE_VALUE, This price definition already exists in this price book: []`

It is like if the framework isn't able to insert new pricebook entries.

The strange thing is that the class seems to work as intended if I write a custom test using Account and Contact objects (literally using two nested loops to insert some accounts with some contacts on each).

Am I missing something here? Maybe something that is not properly configured on the organization?

**Comment by afawcett** [comment]
That is odd, normally the tests are isolated from the org. Which test method is failing?

**Comment by renatoliveira** [comment]
`testUnitOfWorkNewDirtyDelete` and `testDerivedUnitOfWork_CommitSuccess` were failing, while the others were succeeding (3/5).

I've commented these on my test class, and created another two small tests using Account and Contact, which worked just fine.

Here's a gist with the two tests I used: https://gist.github.com/renatoliveira/6f2e02629048b3bc722194f5b43eb6f7

**Comment by afawcett** [comment]
Thanks @renatoliveira this is most odd, since i think your the first to report this. Are you able to confirm if its still the case? Wondering if it was a platform bug thats been fixed? Otherwise, i think we can change the impl to make the repo test class work for you in the future and others if they hit this issue. As such i've marked this as an enhancement.

**Comment by renatoliveira** [comment]
@afawcett this is odd indeed. I've deployed the class and its test successfully on a DE environment right now (as a test), and I didn't have to change anything. When I first described this issue, it was on a sandbox environment, but I don't think there's a difference. I thought that this might be a configuration issue, but I couldn't find anything that could break the code (like a workflow or validation rule, for example).

I managed to deploy it on said org by modifying the test class a little bit though, so it isn't really a problem right now. :) Created the issue just to let you know, and maybe as a reference if someone else gets the same error.
